### PR TITLE
feat(suite): wallet remmember settings: default wallet loading

### LINKED ISSUE #12394: Add default wallet loading to Settings (view-only)
Screenshot 2024-05-15 at 10 14 01

Screenshot 2024-05-17 at 15 39 58

https://www.figma.com/design/wGshbwWuR7shAA1i1UBqR8/branch/1O5PSi8KzMpeVyFMrEMZeA/Remember-Mode?node-id=2354-40396&t=03r75bJYrFjp9AqS-0

### PR DESCRIPTION
**Pull request by peter-sanderson**
Resolves https://github.com/trezor/trezor-suite/issues/12394

image

### CODE REVIEW

File: packages/suite/src/types/suite/index.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ type DeviceAuthenticityAction = ReturnType<
 
 // all actions from all apps used to properly type Dispatch.
 export type Action =
-    | TrezorConnectEvents
+    | TrezorConnectEvents // Todo: This should not be here, actions shall be defined independently from Connect Events (and they shall be mapped onto them)
```
Comment (peter-sanderson) [inline packages/suite/src/types/suite/index.ts:71]: :point_up: :exclamation: :question:

**Review by jvaclavik — APPROVED** [review]

File: suite-common/wallet-core/src/device/deviceReducer.ts [hunk 2]
```diff
@@ -443,7 +457,7 @@ const forget = (draft: State, device: TrezorDevice) => {
         delete draft.devices[index].authFailed;
         draft.devices[index].state = undefined;
         draft.devices[index].walletNumber = undefined;
-        draft.devices[index].useEmptyPassphrase = !device.features.passphrase_protection;
+        // draft.devices[index].useEmptyPassphrase = !device.features.passphrase_protection;
```
Comment (peter-sanderson) [inline suite-common/wallet-core/src/device/deviceReducer.ts:466]: dont forget to resolve this

File: packages/suite/src/components/suite/layouts/SuiteLayout/useAppShortcuts.tsx [hunk 3]
```diff
@@ -12,8 +12,8 @@ export const useAppShortcuts = () => {
 
         // press CMD + P to show PassphraseModal
         if (modKey && e.key === 'p' && device) {
-            dispatch(createDeviceInstance({ device }));
-            e.preventDefault(); // prevent default behaviour
+            dispatch(createDeviceInstance({ device, useEmptyPassphrase: false }));
+            e.preventDefault(); // prevent default behavior
```
Comment (jvaclavik) [inline packages/suite/src/components/suite/layouts/SuiteLayout/useAppShortcuts.tsx:16]: Actually the previous one was correct

File: packages/suite/src/components/suite/layouts/SuiteLayout/useAppShortcuts.tsx [hunk 4]
```diff
@@ -25,7 +25,7 @@ export const useAppShortcuts = () => {
                     },
                 }),
             );
-            e.preventDefault(); // prevent default behaviour
+            e.preventDefault(); // prevent default behavior
```
Comment (jvaclavik) [inline packages/suite/src/components/suite/layouts/SuiteLayout/useAppShortcuts.tsx:28]: same here :D

File: packages/suite/src/views/suite/SwitchDevice/DeviceItem/DeviceItem.tsx [hunk 5]
```diff
@@ -65,16 +71,41 @@ export const DeviceItem = ({ device, instances, onCancel, backgroundRoute }: Dev
...
-    const selectDeviceInstance = (instance: DeviceItemProps['device']) => {
-        dispatch(selectDeviceThunk(instance));
+    const selectDeviceInstance = ({
+        device,
+        walletType,
+    }: {
+        device: DeviceItemProps['device'];
+        walletType?: WalletType;
+    }) => {
+        console.log('walletType', walletType);
```
Comment (jvaclavik) [inline packages/suite/src/views/suite/SwitchDevice/DeviceItem/DeviceItem.tsx]: delete console log 🙏

**Review by jvaclavik — COMMENTED** [review]
Looks good!

File: packages/suite/src/components/suite/layouts/SuiteLayout/useAppShortcuts.tsx (hunk 3 above)
Comment (peter-sanderson) [reply packages/suite/src/components/suite/layouts/SuiteLayout/useAppShortcuts.tsx:16]: image

This is what spell-checked in VSCode wants :D

:grin:

File: suite-common/wallet-core/src/device/deviceReducer.ts [hunk 6]
```diff
@@ -501,12 +516,15 @@ export const setDeviceAuthenticity = (
...
-            connectDevice(state, payload);
+        .addCase(deviceActions.connectDevice, (state, { payload: { device, settings } }) => {
+            connectDevice(state, device, settings);
         })
+        .addCase(
+            deviceActions.connectUnacquiredDevice,
+            (state, { payload: { device, settings } }) => {
+                connectDevice(state, device, settings);
+            },
+        )
```
Comment (juriczech) [inline suite-common/wallet-core/src/device/deviceReducer.ts]:
```suggestion
        .addCase(isAnyOf(deviceActions.connectDevice,  deviceActions.connectUnacquiredDevice), (state, { payload: { device, settings } }) => {
            connectDevice(state, device, settings);
        })
```

File: suite-common/wallet-core/src/device/deviceThunks.ts [hunk 7]
```diff
@@ -598,3 +596,27 @@ export const onPassphraseSubmit = createThunk(
         });
     },
 );
+
+type DeviceConnectThunkEventType = typeof DEVICE.CONNECT | typeof DEVICE.CONNECT_UNACQUIRED;
+
+const deviceConnectThunksMap: Record<
+    DeviceConnectThunkEventType,
+    (payload: DeviceConnectActionPayload) => any
```
Comment (juriczech) [inline suite-common/wallet-core/src/device/deviceThunks.ts]:
```suggestion
    (payload: DeviceConnectActionPayload) => AnyAction

File: suite-common/wallet-core/src/device/deviceThunks.ts [hunk 8]
```diff
@@ -598,3 +596,27 @@ export const onPassphraseSubmit = createThunk(
...
+> = {
+    [DEVICE.CONNECT]: deviceActions.connectDevice,
+    [DEVICE.CONNECT_UNACQUIRED]: deviceActions.connectDevice,
+};
+
+type DeviceConnectThunksParams = {
+    type: DeviceConnectThunkEventType;
```
Comment (juriczech) [inline suite-common/wallet-core/src/device/deviceThunks.ts:629]: I believe `type` name is confusing here. How about `connectThunkEventType` or something like that?

File: suite-common/wallet-core/src/device/deviceActions.ts [hunk 9]
```diff
@@ -2,15 +2,26 @@ import { createAction } from '@reduxjs/toolkit';
...
 import { ButtonRequest, TrezorDevice } from '@suite-common/suite-types';
+import { WalletType } from '@suite-common/wallet-types';
 
 export const DEVICE_MODULE_PREFIX = '@suite/device';
 
-const connectDevice = createAction(DEVICE.CONNECT, (payload: Device) => ({ payload }));
+export type ConnectDeviceSettings = {
+    defaultWalletLoading: WalletType;
+    isViewOnlyModeVisible: boolean;
+};
```
Comment (juriczech) [inline suite-common/wallet-core/src/device/deviceActions.ts:12]: I'd move this to @suite-common/wallet-types

**Review by juriczech — COMMENTED** [review]
LGTM, just some nits

File: suite-common/wallet-core/src/device/deviceReducer.ts (hunk 6 above)
Comment (peter-sanderson) [reply suite-common/wallet-core/src/device/deviceReducer.ts]: :white_check_mark:

File: suite-common/wallet-core/src/device/deviceThunks.ts (hunk 7 above)
Comment (peter-sanderson) [reply suite-common/wallet-core/src/device/deviceThunks.ts]: :white_check_mark:

File: suite-common/wallet-core/src/device/deviceThunks.ts (hunk 8 above)
Comment (peter-sanderson) [reply suite-common/wallet-core/src/device/deviceThunks.ts:629]: Cant change it it is a type from action (`action.type`), the result is an action

File: suite-common/wallet-core/src/device/deviceActions.ts (hunk 9 above)
Comment (peter-sanderson) [reply suite-common/wallet-core/src/device/deviceActions.ts:12]: This is basically a payload type of the action, I think it shall be here next to the action it types

**Pull request merged**